File: Remove-Outermost-Parentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string s) 
    {
        string res = "";
        int count = 0;

        for (int i = 0; i < s.size(); i++) 
        {
            if (s[i] == '(') 
            {
                if (count > 0) 
                {
                    res += s[i];  
                }
                count++;
            } 
            else 
            {
                count--;
                if (count > 0) 
                {
                    res += s[i];  
                }
            }
        }

        return res;
    }
};

File: Remove-Outermost-Parentheses.py
class Solution(object):
    def removeOuterParentheses(self, s):
        """
        :type s: str
        :rtype: str
        """
        res = ""
        count = 0
        x = len(s)

        for i in range(x):
            if s[i] == '(':
                if count > 0:
                    res += s[i]  
                count += 1 
            else:
                count -= 1
                if count > 0:
                    res += s[i]

        return res
